src/ui/webeiddialog.cpp

```diff
@@ -399,20 +437,28 @@ bool WebEidDialog::event(QEvent* event)
 {
     if (event->type() == QEvent::LanguageChange) {
         ui->retranslateUi(this);
-        ui->aboutVersion->setText(tr("Version: %1").arg(qApp->applicationVersion()));
+        emit languageChange();
     }
     return WebEidUI::event(event);
 }
 
-void WebEidDialog::onRetryImpl(const QString& error)
+void WebEidDialog::onRetryImpl(const std::function<QString()>& text)
 {
-    ui->connectCardLabel->setText(error);
-    ui->messagePageTitleLabel->setText(tr("Operation failed"));
-    ui->cardChipIcon->setPixmap(pixmap(QLatin1String("no-id-card")));
-    setupOK([this] { emit retry(); }, tr("Try again"), true);
+    setTRText(ui->connectCardLabel, text);
+    setTRText(ui->messagePageTitleLabel, [] { return tr("Operation failed"); });
+    ui->cardChipIcon->setPixmap(pixmap("no-id-card"_L1));
+    setupOK([this] { emit retry(); }, [] { return tr("Try again"); }, true);
     ui->pageStack->setCurrentIndex(int(Page::ALERT));
 }
 
+void WebEidDialog::setTRText(QWidget* label, const std::function<QString()>& text)
```

Comment:
I really like this! Simple and elegant solution to the problem. 

Can I suggest a slightly more readable method name though? The name should answer the question _What is it?_ or _What does it do?_ and to my mind, the answer would be _Dynamic translation_. 

```suggestion
void WebEidDialog::dynamicTranslate(QWidget* label, const std::function<QString()>& text)
```

Or maybe there is even a better name?

Reply:
maybe setTextTR? It says modifies setText property using tr() function.